driver/src/contracts/stablex_auction_element.rs

```diff
@@ -0,0 +1,227 @@
+        let auction_element = StableXAuctionElement {
+            valid_from: U256::from(2),
+            valid_until: U256::from(261),
+            sell_token_balance: 3,
+            order: Order {
+                batch_information: Some(BatchInformation {
+                    slot_index: 0,
+                    slot: U256::from(0),
+                }),
+                account_id: H160::from(1),
+                buy_token: 258,
+                sell_token: 257,
+                buy_amount: 258 * 257 / 259,
+                sell_amount: 257,
+            },
+        };
+        assert_eq!(res, auction_element);
+    }
+    #[test]
+    fn custom_auction_element_from_bytes_with_higher_order_id() {
+        let bytes: [u8; 113] = [
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, // user: 20 elements
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+            0, 0, 3, // sellTokenBalance: 3, 32 elements
+            1, 2, // buyToken: 256+2,
+            1, 1, // sellToken: 256+1, 56
+            0, 0, 0, 2, // validFrom: 2
+            0, 0, 1, 5, // validUntil: 256+5 64
+            1, // is_sell_order: true
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 2, // priceNumerator: 258
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 3, // priceDenominator: 259
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, // remainingAmount: 2**8 + 1 = 257
+        ];
+        let mut order_count = HashMap::new();
+        StableXAuctionElement::from_bytes(&mut order_count, &bytes);
+        let mut bytes_modified = bytes.clone();
+        bytes_modified[112] = 0; // setting remainingAmount: 2**8  = 256
+        let res = StableXAuctionElement::from_bytes(&mut order_count, &bytes_modified);
+        let auction_element = StableXAuctionElement {
+            valid_from: U256::from(2),
+            valid_until: U256::from(261),
+            sell_token_balance: 3,
+            order: Order {
+                batch_information: Some(BatchInformation {
+                    slot_index: 1,
+                    slot: U256::from(0),
+                }),
+                account_id: H160::from(1),
+                buy_token: 258,
+                sell_token: 257,
+                buy_amount: 258 * 256 / 259,
+                sell_amount: 256,
+            },
+        };
+        assert_eq!(res, auction_element);
+    }
+
+    #[test]
+    #[should_panic]
+    fn test_from_bytes_fails_on_hopefully_null() {
+        let mut order_count = HashMap::new();
+        StableXAuctionElement::from_bytes(&mut order_count, &[1u8; 113]);
+    }
+
+    // Testing in_auction
+    #[test]
+    fn in_auction_left() {
+        let mut element = StableXAuctionElement::default();
+        element.valid_from = U256::from(2);
+        element.valid_until = U256::from(5);
+        assert_eq!(element.in_auction(U256::from(2)), true);
```

Comment:
This will have to be changed when the `<=` is introduced.

Reply:
It was already changed in my PR. Otherwise, tests would not have passed :). Maybe you are again looking at old prs?